tests/api/test_views.py

```diff
@@ -582,6 +620,132 @@ def test_post_enterprise_customer_user_logged_out(self):
+                "Unable to fetch API response for catalog courses from endpoint "
+                "'/enterprise-customer/d2098bfb-2c78-44f1-9eb2-b94475356a3f/courses'."
+            )},
+        ),
+        (
+            'd2098bfb-2c78-44f1-9eb2-b94475356a3f',
+            1,
+            reverse('enterprise-customer-courses', ('d2098bfb-2c78-44f1-9eb2-b94475356a3f',)),
+            True,
+            MOCK_CATALOG_COURSE_RESPONSE,
+            {
+                'count': 3,
+                'next': ('http://testserver/enterprise/api/v1/enterprise-customer/'
+                         'd2098bfb-2c78-44f1-9eb2-b94475356a3f/courses/?page=3'),
+                'previous': ('http://testserver/enterprise/api/v1/enterprise-customer/'
+                             'd2098bfb-2c78-44f1-9eb2-b94475356a3f/courses/?page=1'),
+                'results': [
+                    {
+                        'owners': [
+                            {
+                                'description': None,
+                                'tags': [],
+                                'name': '',
+                                'homepage_url': None,
+                                'key': 'edX',
+                                'certificate_logo_image_url': None,
+                                'marketing_url': None,
+                                'logo_image_url': None,
+                                'uuid': 'aa4aaad0-2ff0-44ce-95e5-1121d02f3b27'
+                            }
+                        ],
+                        'tpa_hint': None,
+                        'catalog_id': 1,
+                        'enterprise_id': 'd2098bfb-2c78-44f1-9eb2-b94475356a3f',
+                        'uuid': 'd2fb4cb0-b538-4934-ba60-684d48ff5865',
+                        'title': 'edX Demonstration Course',
+                        'prerequisites': [],
+                        'image': None,
+                        'expected_learning_items': [],
+                        'sponsors': [],
+                        'modified': '2017-03-03T07:34:19.322916Z',
+                        'full_description': None,
+                        'subjects': [],
+                        'video': None,
+                        'key': 'edX+DemoX',
+                        'short_description': None,
+                        'marketing_url': None,
+                        'level_type': None,
+                        'course_runs': []
+                    }
+                ]
+            }
+        ),
+    )
+    @ddt.unpack
+    @mock.patch('enterprise.api.v1.views.CourseCatalogApiClient')
+    def test_enterprise_customer_courses(
+            self,
+            enterprise_uuid,
+            catalog,
+            url,
+            link_user,
+            mocked_catalog_courses,
+            expected,
+            mock_catalog_api_client
+    ):
+        """
+        Make sure if an enterprise does not exist for this call, we return a descriptive error.
```

Comment:
@brittneyexline I think we should update this to reflect the fact that this test checks multiple scenarios; right now it sounds like the test is only verifying a single scenario.